java.source.base/src/org/netbeans/modules/java/source/usages/BuildArtifactMapperImpl.java

```diff
@@ -302,30 +313,31 @@ private static void copyFile(FileObject updatedFile, File target) throws IOExcep
         }
     }
 
-    private static void copyRecursively(File source, File target) throws IOException {
-        if (source.isDirectory()) {
-            if (target.exists() && !target.isDirectory()) {
-                throw new IOException("Cannot create folder: " + target.getAbsolutePath() + ", already exists as a file.");
-            }
+    private static void copyRecursively(File source, File target, URL sourceURL) throws IOException {
+        if (target.exists() && !target.isDirectory()) {
+            throw new IOException("Cannot create folder: " + target.getAbsolutePath() + ", already exists as a file.");
+        }
 
-            File[] listed = source.listFiles();
-            
-            if (listed == null) {
-                return ;
-            }
-            
-            for (File f : listed) {
-                String name = f.getName();
-                if (name.endsWith(SIG))
-                    name = name.substring(0, name.length() - FileObjects.SIG.length()) + FileObjects.CLASS;
-                copyRecursively(f, new File(target, name));
-            }
-        } else {
-            if (target.isDirectory()) {
-                throw new IOException("Cannot create file: " + target.getAbsolutePath() + ", already exists as a folder.");
-            }
+        File[] listed = source.listFiles();
 
-            copyFile(source, target);
+        if (listed == null) {
+            return ;
+        }
+
+        for (File f : listed) {
+            String name = f.getName();
+            if (name.endsWith(SIG))
+                name = name.substring(0, name.length() - FileObjects.SIG.length()) + FileObjects.CLASS;
```

Comment:
You are already doing something with .sig here...

Reply:
This is the same as the original line 320. [Reduced diff](https://github.com/apache/incubator-netbeans/pull/504/files?w=1)